src/hooks/useSingleStageHandler.ts

```diff
@@ -258,6 +260,32 @@ export const useSingleStageHandler = () => {
     }
   }, [address, provider, stageAddress, state]);
 
+  useEffect(() => {
+    const fetchStageData = async () => {
+      if (DAOSEEDER_FACTORY_ADDRESS && campaignId) {
+        const contract = await getSmartContractWithProvider(
+          DAOSEEDER_FACTORY_ADDRESS,
+          provider,
+          JSON.stringify(DaoSeederFactory.abi)
+        );
+        const data = await getCampaign(campaignId, contract);
+        if (data && parseInt(data.stageCount.toString()) > 0) {
+          for (let i = 1; i <= parseInt(data.stageCount.toString()); i++) {
+            const stageKey = await getStageKey(data.tokenAddress, i);
+            if (stageKey === stageId) {
+              setCampaignTitle(data.name + " - Stage (" + i + ")");
```

Comment:
I think this should be done in fetchStage line 89, not a separate loop

Reply:
The loop is there because we need to check which stage number is the current stage in the campaign. In fetchStage we do not have this information